Scripts/parallel_mtweedietests_method.R

```diff
@@ -0,0 +1,65 @@
+# Libraries
+library(tweedie)
+library(ggplot2)
+library(parallel)
+library(magrittr)
+library(tidyverse)
+
+# Functions
+simTweedieTest <- function(N) { 
+  t.test(rtweedie(N, mu=10000, phi=100, power=1.9), mu=10000)$p.value 
+}
+
+# Parallel MTweedieTests
+MTweedieTests_parallel <- function(N, M, sig) {
+  num_cores <- detectCores() - 1
+  results <- mclapply(1:M, function(x) simTweedieTest(N), mc.cores=num_cores)
+  sum(unlist(results) < sig) / M
+}
+
+# Non-parallel MTweedieTests
+MTweedieTests <- function(N, M, sig) { 
+  sum(replicate(M, simTweedieTest(N)) < sig) / M 
+}
+
+df <- expand.grid(N = c(10,100,1000,5000, 10000), M = 1000, share_reject = NA) 
+
+# Using parallelized function for Assignment 3
+num_cores <- detectCores() - 1
+compute_share_reject <- function(i) {
+  MTweedieTests_parallel(N = df$N[i], M = df$M[i], sig = .05)
+}
```

Comment:
You should not have used parallization here. This is because when you run this line below, all cores are used on this function and not on the MTweedieTests_parallel() function. Therefore you will not utilize several cores on the M simulations